development-tools/clinic.rst

```diff
@@ -2001,51 +2001,72 @@ The generated glue code looks like this:
 
 
 .. _clinic-howto-getter:
+.. _clinic-howto-getter-setter:
 
 How to generate a getter
 ------------------------
 
-"Getters" are C functions that facilitate property-like access for a class.
-See :c:type:`getter <PyGetSetDef>` for details.
-You can use the ``@getter`` directive to generate an "impl" function for a
+"Getters" and "setters" are C functions that facilitate property-like access for a class.
+See :c:type:`getter <PyGetSetDef>` and :c:type:`setter <PyGetSetDef>` for details.
+You can use the ``@getter`` and ``@setter`` directive to generate an "impl" function for a
 getter using Argument Clinic.
 
-This example -- taken from :cpy-file:`Modules/_io/bufferedio.c` --
-shows the use of ``@getter`` in combination with
+This example -- taken from :cpy-file:`Modules/_io/textio.c` --
+shows the use of ``@getter`` and ``@setter`` in combination with
 the :ref:`@critical_section <clinic-howto-critical-sections>` directive
 (which achieves thread safety without causing deadlocks between threads)::
 
     /*[clinic input]
     @critical_section
     @getter
-    _io._Buffered.closed
+    _io.TextIOWrapper._CHUNK_SIZE
+    [clinic start generated code]*/
+
+    /*[clinic input]
+    @critical_section
+    @setter
+    _io.TextIOWrapper._CHUNK_SIZE
     [clinic start generated code]*/
 
 The generated glue code looks like this:
 
 .. code-block:: c
 
     static PyObject *
-    _io__Buffered_closed_get(buffered *self, void *context)
+    _io_TextIOWrapper__CHUNK_SIZE_get(textio *self, void *Py_UNUSED(context))
     {
         PyObject *return_value = NULL;
 
         Py_BEGIN_CRITICAL_SECTION(self);
-        return_value = _io__Buffered_closed_get_impl(self);
+        return_value = _io_TextIOWrapper__CHUNK_SIZE_get_impl(self);
         Py_END_CRITICAL_SECTION();
 
         return return_value;
     }
 
+    static int
+    _io_TextIOWrapper__CHUNK_SIZE_set(textio *self, PyObject *value, void *Py_UNUSED(context))
+    {
+        int return_value;
+        Py_BEGIN_CRITICAL_SECTION(self);
+        return_value = _io_TextIOWrapper__CHUNK_SIZE_set_impl(self, value);
+        Py_END_CRITICAL_SECTION();
+        return return_value;
+    }
+
+Note that each of ``@getter`` and ``@setter`` should be declared in separated definitions.
 And then the implementation will work the same as a Python method which is
 decorated by :py:class:`property`.
```

Comment:
```suggestion
The implementation will then work the same as a Python method which is
decorated by :py:class:`property`.
Note that each of ``@getter`` and ``@setter`` should be declared in separated definitions.
```
Starting a sentence with "And" sounds a bit weird, especially with a note between the example and the sentence.  I'd move the note after the sentence, and drop the "And" anyway.